content/protocols/sse.textile

```diff
@@ -0,0 +1,134 @@
+
+h4(#detecting-token-expiry). Detecting token expiry
+
+When a connection is closed as a result of any error (that is, it's not just a dropped connection), then the @error@ event will occur on the @EventSource@ instance, and the data attribute of the event will contain an Ably error body with the information about the nature of error. In the case of a token error - that is an error arising from a problem with the auth token - the code in the error body will indicate that. Token errors have a code in the range @40140 <= code < 40150@. In such cases, the authentication can be retried with a new @accessToken@.
+
+In the future we plan to send an event on the connection that indicates that the token will expire imminently, which will allow a new connection to be established prior to the closure of the previous connection.
+
+h4(#specifying-lasteventid). Specifying the lastEventId
+
+Each message received will have a @lastEventId@ attribute containing the last id of any message received on the connection. When constructing a new connection, this value can be specified as a @lastEvent@ param in the URL.
+
+The following example of implementing message continuity with token auth:
+
+```[javascript]
+let lastEvent;
+
+const connectToAbly = () => {
+  // obtain a token
+  const token = <GET-NEW-ABLY-AUTH-TOKEN>
+
+  // establish a connection with that token
+  const lastEventParam = lastEvent ? ('&lastEvent=' + lastEvent) : '';
+  eventSource = new EventSource(`https://realtime.ably.io/sse?v=1.1&accessToken=${token}&channels=${channel}${lastEventParam}`);
+
+  // handle incoming messages
+  eventSource.onmessage = msg => {
+    lastEvent = msg.lastEventId;
+    // ... normal message processing
+  }
+
+  // handle connection errors
+  eventSource.onerror = msg => {
+    const err = JSON.parse(msg.data);
+    const isTokenErr = err.code >= 40140 && err.code < 40150;
+    if(isTokenErr) {
+      eventSource.close();
+      connectToAbly();
+    } else {
+      // ... handle other types of error -- for example, retry on 5xxxx, close on 4xxxx
+    }
+  }
+}
+
+connectToAbly();
+```
+
+An important thing to note here is that the EventSource API tries to auto-reconnect and re-subscribe to the SSE endpoint when any error occurs, even the token expiry error like in this case. This means that upon manually re-subscribing to the SSE endpoint with a new token, there will be two active subscriptions to the endpoint - one with the old token which would continue to throw an error due to expired credentials and another with the new token. Hence, it is important to close the previous @EventSource@ subscription using @eventSource.close()@ before re-subscribing with the new token as shown in the snippet above.
+
+You can take a look at a "demo app":https://sse-token-auth.glitch.me and "a complete code example for implementing message continuity in an SSE subscription when using token auth":https://glitch.com/edit/#!/sse-token-auth.
+
+h2(#channel-options). Channel options
```

Comment:
I think this section is still missing some of the information from [this](https://ably.com/docs/realtime/channels/channel-parameters/deltas?lang=javascript#using-deltas-non-ably) page relevant to SSE, as well as [this one](https://ably.com/docs/realtime/channels/channel-parameters/overview?lang=javascript#using-parameters-non-ably).